### ISSUE TITLE
Calculate average for overlapping parts when merging rasters

### ISSUE THREAD

**Issue by magnusnil** [qgis/QGIS#46990]
### Feature description

When merging rasters using the Merge tool in processing, there are no options for how to handle overlapping parts. In my case, the files overlap with a few hundred meters, and for these parts I would like QGIS to calculate the average value for the raster files (2 or more files). See for example how FME allows this kind of manipulation, http://docs.safe.com/fme/html/FME_Desktop_Documentation/FME_Transformers/Transformers/rastermosaicker.htm.

**Comment by gioman** [comment]
@magnusnil  I agree is a very common need when merging rasters. Anyway if we are talking about the tool in processing > gdal > miscellaneous > merge, then this feature request should probably be filed here https://github.com/OSGeo/gdal/issues (the tool comes from GDAL is not a QGIS core one).

**Comment by magnusnil** [comment]
Thanks! I created a GDAL issue for this.
